apps/users/views.py

```diff
@@ -73,6 +81,13 @@ def profile(request):
         elif "save_preferences" in request.POST:
             if profile_form.is_valid():
                 profile_form.save()
+
+                # Subscribe user to newsletter when consented
+                if profile_form.cleaned_data.get("subscribed_newsletter"):
+                    add_subscription(request.user.profile)
+                else:
+                    remove_subscription(request.user.profile)
+
```

Comment:
Does this happen async? And is it maybe more efficient to first check if they changed this specific preference before sending an api request to sendgrid?

Reply:
it is async. I have handled it in a later commit by referencing to old instance of model